src/tools/search-variants-mapi.ts

```diff
@@ -0,0 +1,137 @@
+import type { McpServer } from "@modelcontextprotocol/sdk/server/mcp.js";
+import pRetry from "p-retry";
+import { createMapiClient } from "../clients/kontentClients.js";
+import type { AppConfiguration } from "../config/appConfiguration.js";
+import { searchOperationSchema } from "../schemas/searchOperationSchemas.js";
+import { handleMcpToolError } from "../utils/errorHandler.js";
+import { createMcpToolSuccessResponse } from "../utils/responseHelper.js";
+
+interface AiOperationResponse {
+  operationId: string;
+}
+
+interface AiOperationResultResponse {
+  message: string;
+  operationId: string;
+  result?: any;
```

Comment:
Can we have a better type here? The least you can do is use `unknown` instead and ideally you would have a precise type of what you expect there and have zod parse it.